src/main/java/me/arminb/sara/dao/UserDAOImpl.java

```diff
@@ -0,0 +1,154 @@
+package me.arminb.sara.dao;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.mongodb.*;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.UpdateResult;
+import me.arminb.sara.entities.User;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository("userDAO")
+public class UserDAOImpl implements UserDAO {
+
+    Logger logger = LoggerFactory.getLogger(UserDAOImpl.class);
+
+    @Autowired
+    private MongoDatabase database;
+
+    @Override
+    public List<User> findAll() throws DataAccessException {
+        List<User> list = new ArrayList();
+        try {
+            MongoCollection<Document> collection = database.getCollection("users");
+            MongoCursor<Document> iterator = collection.find().iterator();
+            while (iterator.hasNext()) {
+                String jsonInString = iterator.next().toJson();
```

Comment:
use Jackson ObjectMapper instead of Gson.
`ObjectMapper objectMapper = new ObjectMapper();`
`User user = objectMapper.readValue(jsonInString, User.class)`